src/server/v3/resources/images.py

```diff
@@ -373,18 +374,36 @@ def patch(self, image_id):
                         current_app.logger.info(f"The artifact {value} is not in S3 and "
                                                 f"was not soft-deleted. Ignoring")
                         current_app.logger.info(str(exc))
+                setattr(image, key, value)
             elif key == "arch":
                 current_app.logger.info(f"Patching architecture with {value}")
                 image.arch = value
+                setattr(image, key, value)
+            elif key == 'metadata':
+                operation = value.get('operation')
+                annotation_key = value.get('key')
+                annotation_value = value.get('value', '')
+                current_app.logger.debug("Image Patch changeset: Current: %s -> %s %s %s"
+                                        % (image.metadata, operation, annotation_key, annotation_value))
+                if operation not in ['set', 'remove']:
+                    current_app.logger.info(f"Unknown requested operation change '{operation}'.")
+                    return generate_data_validation_failure(errors=[])
+                if operation == 'set':
+                    image.metadata[annotation_key] = annotation_value
+                elif operation == 'remove':
+                    try:
+                        del image.metadata[annotation_key]
+                    except KeyError:
+                        current_app.logger.info("No-op when removing non-existent metadata from IMS record.")
+                current_app.logger.debug("Image metadata result: %s", image.metadata)
             else:
                 current_app.logger.info(f"{log_id} Not able to patch record field {key} with value {value}")
                 return generate_data_validation_failure(errors=[])
+        current_app.logger.info(f"{log_id} image metadata information dump: '%s'" % image.metadata)
```

Comment:
I don't think we need to dump the metadata result after every image patch operation - either dump the entire object (on debug log) or remove this line.